src/uct/ib/dc/dc_mlx5_devx.c

```diff
@@ -141,8 +141,12 @@ ucs_status_t uct_dc_mlx5_iface_devx_dci_connect(uct_dc_mlx5_iface_t *iface,
     if (uct_ib_iface_is_roce(&iface->super.super.super)) {
         UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.eth_prio,
                           iface->super.super.super.config.sl);
-        uct_ib_mlx5_devx_set_qpc_port_affinity(md, path_index, qpc,
-                                               &opt_param_mask);
+        if ((iface->tx.port_select == UCT_DC_MLX5_LAG_PORT_AFFINITY) ||
+            (iface->tx.port_select == UCT_DC_MLX5_LAG_PORT_AUTO &&
+             md->super.dev.port_select_mode == 0)) {
```

Comment:
need to resolve prio on initialization and do only 1 boolean op test here

Reply:
@evgeny-leksikov Do you mean using ```iface->tx.port_select``` as only 1 boolean op test here and set ```iface->tx.port_select``` value after considering the two conditions here?

Reply:
see https://github.com/openucx/ucx/pull/8382/files#r925310697
the result should be calculated on iface initialization stage